app/controllers/follows_controller.rb

```diff
@@ -68,7 +68,11 @@ def follow_params
   end
 
   def follow(followable, need_notification: false)
```

Comment:
`def` is an implicit `begin`, so this could be:

```ruby
def follow(followable, need_notification: false)
  user_follow = current_user.follow(followable)
rescue ActiveRecord::RecordInvalid => e
  Rails.logger.error(e)
end
```

See e.g. https://www.rubyguides.com/2019/06/ruby-rescue-exceptions/

Reply:
Thank you @citizen428 . I am seeing that codeclimate/diff-coverage check failed. How can I do to turn this check in green?

Reply:
unfortunately the report is not very forthcoming on details, so I think you can ignore it :)

Reply:
@citizen428 If I don't use begin keyword then I have to use return statement  in ensure block. Is this ok for you?
```
def follow(followable, need_notification: false)
  user_follow = current_user.follow(followable)
rescue ActiveRecord::RecordInvalid => e
  DataDogStatsClient.increment("users.invalid_follow")
ensure
   Notification.send_new_follower_notification(user_follow) if need_notification
   return "followed"
end
```

Reply:
I would set it up like this and actually I think that reads pretty cleanly personally. We will see what @citizen428 thinks. 
```ruby
def follow(followable, need_notification: false)
  user_follow = current_user.follow(followable)
  Notification.send_new_follower_notification(user_follow) if need_notification
rescue ActiveRecord::RecordInvalid => e
  DataDogStatsClient.increment("users.invalid_follow")
ensure
   return "followed"
end
```

Reply:
I'm not sure if we should really return "followed" in case something went wrong. Maybe just have different returns based on whether we succeed or fail?

```ruby
def follow(followable, need_notification: false)
  user_follow = current_user.follow(followable)
  Notification.send_new_follower_notification(user_follow) if need_notification
  "followed"
rescue ActiveRecord::RecordInvalid => e
  DataDogStatsClient.increment("users.invalid_follow")
  # some other message, e.g. "previously followed"
end
```

Either way, it's such a small detail I'll leave it to your judgement what you think is best, i.e. if you want to keep as-is, I have no problem with that.

Reply:
Thank you folks for your help. I ended up removing `ensure` because Rubocop doesn't like to have a `return` within `ensure`.